source/common/network/connection_impl.cc

```diff
@@ -757,7 +757,14 @@ void ClientConnectionImpl::connect() {
     ASSERT(connecting_);
   } else {
     ASSERT(result.rc_ == -1);
-    if (result.errno_ == EINPROGRESS) {
+#ifdef WIN32
```

Comment:
This is an example of an error code that is different on Windows vs POSIX, personally I think it may be valuable to keep the #ifdef to ensure developers understand the differences between `connect()` on different platforms, but we could potentially abstract this error check in a different way, open to feedback about this

Reply:
I like the approach as long as we don't have too many instances of the same syscall